packages/execution-environments/src/common/endowments/index.test.ts

```diff
@@ -219,6 +222,65 @@ describe('Endowment utils', () => {
       teardown();
       jest.runAllTimers();
     });
+
+    it('should not expose then or catch after teardown has been called', async () => {
+      fetchMock.mockOnce('Resolved');
+
+      const mockWallet = { foo: Symbol('bar') };
+      const { endowments, teardown } = createEndowments(mockWallet as any, [
+        'fetch',
+      ]);
+      const { fetch } = endowments as {
+        fetch: typeof globalThis.fetch;
+      };
+      const ErrorProxy = jest
+        .fn()
+        .mockImplementation((reason) => Error(reason));
+
+      const errorNameOne = 'SHOULD_NOT_BE_REACHED_1';
+      const errorNameTwo = 'SHOULD_NOT_BE_REACHED_2';
+      // eslint-disable-next-line jest/valid-expect-in-promise
+      fetch('foo.com')
```

Comment:
I've already been trying to achieve something like that, but if I do so, the tests timeout because the `await teardown()` never finish.

Here is implementation of that test, just changed a little bit because of the TypeScript rules:
```typescript
it('should not expose then or catch after teardown has been called', async () => {
  let fetchResolve: ((result: string) => void) | null = null;
  fetchMock.mockOnce(() => new Promise((r) => (fetchResolve = r)));

  const { fetch, teardownFunction } = network.factory();
  const ErrorProxy = jest
    .fn()
    .mockImplementation((reason) => Error(reason));

  // eslint-disable-next-line jest/valid-expect-in-promise
  fetch('foo.com')
    .then(() => {
      throw new ErrorProxy('SHOULD_NOT_BE_REACHED');
    })
    .catch(() => {
      throw new ErrorProxy('SHOULD_NOT_BE_REACHED');
    })
    .catch((e) => console.log(e));

  await teardownFunction();
  (fetchResolve as any)('Resolved');
  await new Promise((r) => setTimeout(() => r('Resolved'), 0));

  expect(ErrorProxy).not.toHaveBeenCalled();
});
```
So, what happens in the process is that in this function:
```typescript
const teardownFunction = async () => {
    teardownRef.lastTeardown += 1;
    const promises: Promise<void>[] = [];
    openConnections.forEach(({ cancel }) => promises.push(cancel()));
    openConnections.clear();
    await Promise.all(promises);
  };
```
The last step `await Promise.all(promises);` never finish.

Reply:
The teardown never finishes because the `cancel()` function waits for the fetch to resolve/reject, which it never does because we resolve it after we finish the teardown.

The `abortController.abort()` inside the `cancel()` should reject the fetch though, I'm not sure why it doesn't. You would have to look into details of fetch mock in how it is implemented.


The below should side-step the problem for this test, but understanding why the `abortController.cancel()` doesn't reject is important and should be researched.
```typescript
const teardownPromise = teardownFunction();
(fetchResolve as any)('Resolved');
await teardownPromise;
```

Reply:
Ok, added side-step solution you mentioned. It works that way.